### DISCUSSION TITLE
How to install package from self hosted GitLab repo using HTTPS

### DISCUSSION THREAD

**Discussion by mawilms** [python-poetry/poetry#8163] [Q&A] [answered]
Hello everyone,

I have a package that I usually install like this:

git+https://${MY_USER}:${MY_PASSWORD}@gitlab.self-hosted.com/group/sub-group/my-package.git#egg=my_package

How can I add this command to the pyproject.toml to be able to install the package?

When I'm using:

`poetry add git+https://${MY_USER}:${MY_PASSWORD}@gitlab.self-hosted.com/group/sub-group/my-package.git#egg=my_package`

I get the following error:

**ssh: Could not resolve hostname git+https**

**Comment by deadnews** [comment]
https://github.com/python-poetry/poetry/issues/835#issuecomment-882695188
https://github.com/python-poetry/poetry/issues/835#issuecomment-1475042301

**Reply by mawilms** [reply]
Hi. Thanks for your answer. Unfortunately I haven't found the solution there.

Right now it is not possible to use GitLab pypi package registry directly. Therefore I have to reference the repo.git address.

**Reply by mawilms** [reply, marked as answer]
Alright it seems that I have searched at the wrong location. If you use a git repo, it seems that poetry reuses the locally stored git credentials. Therefore you don't have to create a source entry in your pyproject.toml
